internal/erofsutils/mount_linux.go

```diff
@@ -43,6 +44,98 @@ func ConvertTarErofs(ctx context.Context, r io.Reader, layerPath string, mkfsExt
 	return nil
 }
 
+// prepareInputFile takes a reader and returns a seekable file along with a flag
+// indicating if the file is temporary and should be cleaned up afterward.
+func prepareInputFile(ctx context.Context, r io.Reader) (file *os.File, isTemp bool, err error) {
+	// Check if r is already a file that we can seek in
+	if f, ok := r.(*os.File); ok {
+		// Seek to the beginning for processing
+		if _, err = f.Seek(0, io.SeekStart); err != nil {
+			return nil, false, fmt.Errorf("failed to seek to beginning of tar file: %w", err)
+		}
+		log.G(ctx).Debug("Using direct file access")
+		return f, false, nil
+	}
+
+	// For regular readers, create a temporary file
+	tempFile, err := os.CreateTemp("", "erofs-tar-*")
+	if err != nil {
+		return nil, false, fmt.Errorf("failed to create temporary tar file: %w", err)
+	}
+
```

Comment:
I think you could just call `os.Remove(tempFile.Name())` in advance unconditionally here.
Using an unlinked inode (or called an orphan inode) is fine since it's just a temporary file.